backend/modules/pdf_processing/ai_extraction_service.py

```
@@ -504,7 +421,17 @@ async def generate_scenes(
                 scenes_json = json_match.group(1)
                 scenes = json.loads(scenes_json)
                 logger.info(f"[SUCCESS] Generated {len(scenes)} scenes")
-                
+
+                # Guard: reject clearly broken outputs before they reach the DB.
+                # GPT-4o should always return 3-6 scenes given the prompt guidance.
+                # Fewer than 3 usually means content was too thin or the model collapsed;
+                # more than 6 usually means it hallucinated extra scenes.
+                if not (3 <= len(scenes) <= 6):
+                    raise ValueError(
+                        f"Scene count out of expected range: got {len(scenes)}, expected 3–6. "
```

Ruff (0.15.12)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)